swap_meet/vendor.py

```diff
@@ -1,2 +1,81 @@
+from swap_meet.item import Item
+from operator import attrgetter
+
+# Wave 1
+#======================================================
 class Vendor:
-    pass
\ No newline at end of file
+    #optionally pass in argument "inventory"
+    def __init__(self, inventory = None):
+        # if "inventory" arugment is passed in, use that
+        if inventory:
+            self.inventory = inventory
+        # otherwise, if there's no inventory argument, initialize an empty list
+        else:
+            self.inventory = []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item 
+        return False
+
+# Wave 2
+#======================================================
+# need to import from Item for this method to work
+    def get_by_category(self, category):
+        inventory_categories = []
+        for item in self.inventory:
+            if item.category == category:
+                inventory_categories.append(item)
+        return inventory_categories
+
+# Wave 3
+#======================================================
+    def swap_items(self, friend_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in friend_vendor.inventory:
+            self.remove(my_item)
+            friend_vendor.add(my_item)
+            friend_vendor.remove(their_item)
+            self.add(their_item)
+            return True
+        else:
+            return False
+
+# Wave 4
+#======================================================
+    def swap_first_item(self, friend_vendor):
+        if not self.inventory or not friend_vendor.inventory:
+            return False        
+        if self.inventory and friend_vendor.inventory:
+            self.swap_items(friend_vendor, self.inventory[0], friend_vendor.inventory[0])
+            return True
```

Comment:
Since `swap_items()` will return a boolean as well, we can shorten these two lines to just one line: 
```
return self.swap_items(friend_vendor, self.inventory[0], friend_vendor.inventory[0])
```